pkg/arvo/neo/cod/std/src/fil/messenger.md

```diff
@@ -562,3 +563,273 @@ The `stem` is head-tagged with a `care:neo`. If the shrub’s dependent is liste
+- `%acked`: acknowledge creation of a DM.
+- `%post`: send a DM.
+
+#### curb:neo
+Cells like `[%pro %ship]` and `[%pro %message-pub]` are examples of `$curb:neo`. This is a powerful type that's beyond the remit of these tutorials, but it's worth clarifying what these cells mean.
+
+```hoon
++$  curb
+  $~  [%pro %$]
+  $%  [%or p=(list curb)]
+      [%only p=stud]
+      [%rol p=stud q=curb]
+      [%not p=curb q=curb]
+      [%pro p=stud]
+      [%any ~]
+  ==
+```
+
+In all of the shrubs we've looked at in these tutorials we could replace every `%pro` curb with the likes of `[%only %ship]` and `[%only %message-pub]` and lose none of the functionality we've looked at. The `[%only %ship]` curb just declares that the state is exclusively a `ship`. However, the `[%pro ship]` curb says that the state can be any type *which can be converted to a `ship` through an available `/con` file*. This has implications for interoperability and state transitions we have not yet fully explored.
+
+#### form:neo
+When `/imp/dm` is first created with a `%make` card, it needs to be created with some pre-defined state. The intial state it accepts has to be a `%dm-diff`. This is a little unusual, but it's essentially the same as a Gall agent sending a poke to itself `+on-init`, except the poke is defined by the external service provider. Since the service provider could be on our ship or someone else's, the `+init` arm takes both `%initiate` and `%invited` pokes. The `+init` arm is the only context in which these pokes are handles.
+
+```hoon
+++  init
+  |=  old=(unit pail:neo)
+  ^-  (quip card:neo pail:neo)
+  ?~  old  !!
+  ?>  =(%dm-diff p.u.old)
+  =/  poke  !<(dm-diff q.u.old)
+  ?+    -.poke  !!
+      :: create me with a pith to a service provider
+      :: to start a new DM with them
+      %initiate
+    :_  ship/!>(partner.poke)
+    :~  :-  (snoc here.bowl %pub)
+        [%make %message-pub ~ ~]
+        ::
+        :-  provider.poke
+        [%poke dm-diff/!>([%invited here.bowl])]
+    ==
+  ::
+      :: create me with a pith to an inviter's dm
+      :: to accept their DM request
+      %invited
+    :_  ship/!>(partner.poke)
+    :~  :-  (snoc here.bowl %pub)
+        [%make %message-pub ~ ~]
+        ::
+        :-  (snoc here.bowl %sub)
+        [%make %message-sub ~ (malt ~[[%pub (snoc dm.poke %pub)]])]
+        ::
+        :-  dm.poke
+        [%poke dm-diff/!>([%acked here.bowl])]
+    ==
+  ==
+```
+
+The `+poke` arm handles `%acked` and `%post` pokes. When we receive an `%acked`, we create an `/imp/message-sub` to subscribe to DMs from the "publisher", which is whoever we're going to talk to. (DM state is symmetrical: both ships are publishing to eachtoher and subscribed to eachother.) When we receive a `%post` from the publisher, we add it to our service provider at `/path/to/this/dm/pub`, which publishes the DM state to the other ship.
```

Comment:
"eachtother" -> "each other"
"eachother" -> "each other"

Reply:
This last sentence shouldn't mention "service provider." The service provider isn't in play at this point anymore. I'd maybe phrase tht last sentence a little differently too; we're not quite publishing it _to_ the other ship, we're just creating it and incidentally the other ship is mirroring it (which on a higher level, yes, is the same as publishing to the other ship).

Could probably stress more in this paragraph that message-sub is still just mirroring messages from the other ship's message-pub.